lib/ansible/executor/task_executor.py

```diff
@@ -450,9 +450,11 @@ def _execute(self, variables=None):
         # the fact that the conditional may specify that the task be skipped due to a
         # variable not being present which would otherwise cause validation to fail
         try:
-            if not self._task.evaluate_conditional(templar, tempvars):
+            conditional_result, failed_condition = self._task.evaluate_conditional_with_result(templar, tempvars)
+            if not conditional_result:
                 display.debug("when evaluation is False, skipping this task")
-                return dict(changed=False, skipped=True, skip_reason='Conditional result was False', _ansible_no_log=no_log)
+                return dict(changed=False, skipped=True, skip_reason='Conditional result was False',
+                            failed_condition=failed_condition, _ansible_no_log=no_log)
```

Comment:
`failed_condition` feels a little too like "something broke"- maybe `false_conditional` or ...

Reply:
That's a good point, I do prefer `false_conditional` here so I'll update it.